etna/datasets/tsdataset.py

```diff
@@ -410,15 +411,41 @@ def _check_endings(self, warning=False):
             else:
                 raise ValueError("All segments should end at the same timestamp")
 
+    def _inverse_transform_target_components(self, target_components_df: pd.DataFrame, target_df: pd.DataFrame):
+        """Inverse transform target components in dataset with inverse transformed target."""
+        if self.target_components_names is not None:
+            self.drop_target_components()
+        inverse_transformed_target_components_df = inverse_transform_target_components(
+            target_components_df=target_components_df,
+            target_df=target_df,
+            inverse_transformed_target_df=self.to_pandas(features=["target"]),
+        )
+        self.add_target_components(target_components_df=inverse_transformed_target_components_df)
+
     def inverse_transform(self, transforms: Sequence["Transform"]):
         """Apply inverse transform method of transforms to the data.
 
         Applied in reversed order.
         """
         # TODO: return regressors after inverse_transform
-        for transform in reversed(transforms):
-            tslogger.log(f"Inverse transform {repr(transform)} is applied to dataset")
-            transform.inverse_transform(self)
+        target_components_present = self.target_components_names is not None
+        target_df, target_components_df = None, None
+        if target_components_present:
```

Comment:
Add a comment that it is done for performance reasons and explain the trick here.